### PR TITLE
Reuse geth if it is already running

### PR DESCRIPTION
**Pull request by paulperegud**

### CODE REVIEW

File: golem/ethereum/node.py [hunk 1]
```diff
@@ -118,6 +132,24 @@ def stop(self):
                 log.warn("Cannot terminate node: process {} no longer exists".format(self.__ps.pid))
 
             self.__ps = None
-            self.rpcport = None
             duration = time.clock() - start_time
             log.info("Node terminated in {:.2f} s".format(duration))
+
+"""
```
Comment (chfast) [inline golem/ethereum/node.py]: The docstring must be inside the function.

File: golem/ethereum/node.py [hunk 2]
```diff
@@ -73,33 +75,45 @@ def __init__(self):
...
+        if self.__ps is not None:
+            raise RuntimeError("Ethereum node already started by us")
+
+        if is_geth_listening(self.testnet):
+            the_chain = "mainnet"
+            if self.testnet:
+                the_chain = "ropsten"
+            running_chain = identify_chain(self.testnet)
+            if running_chain == the_chain:
+                log.info("Using existing Ethereum node ()".format(running_chain))
```
Comment (chfast) [inline golem/ethereum/node.py]: Shouldn't it be `{}`?

**Review by chfast — APPROVED** [review]
Looks good, but 2 small issues.

File: golem/ethereum/node.py (hunk 1 above)
Comment (paulperegud) [reply golem/ethereum/node.py]: fixed

File: golem/ethereum/node.py (hunk 2 above)
Comment (paulperegud) [reply golem/ethereum/node.py]: fixed

File: golem/ethereum/node.py [hunk 3]
```diff
@@ -81,6 +81,9 @@ def __init__(self):
     def is_running(self):
         return self.__ps is not None or self.__system_geth
 
+    def is_reusing(self):
```
Comment (chfast) [inline golem/ethereum/node.py]: Let's make `system_geth` public and that's it.

Comment (paulperegud) [reply golem/ethereum/node.py]: fixed, merging

**Pull request merged**